### PR TITLE
WP-387

### PR DESCRIPTION
**Pull request by paveldhq**

### CODE REVIEW

File: inc/Smartling/Base/SmartlingCoreUploadTrait.php [hunk 1]
```diff
@@ -147,10 +147,65 @@ public function getXMLFiltered(SubmissionEntity $submission)
...
+                if (array_key_exists($_fieldName, $targetContent)) {
+                    $lockedData['entity'][$_fieldName] = $targetContent[$_fieldName];
+                }
+            }
+        }
+
+        return $lockedData;
+    }
+
+    private function arrayMergeIfKeyNotExists($lockedData, $translation)
```
Comment (PavelLoparev) [inline inc/Smartling/Base/SmartlingCoreUploadTrait.php]: Seems like this method should be in `ArrayHelper` class. Can be static.

File: inc/Smartling/Extensions/TranslateLock.php [hunk 2]
```diff
@@ -17,6 +21,79 @@
...
+    {
+        $this->contentHelper = $contentHelper;
+    }
+
+    /**
+     * @return LoggerInterface
+     */
+    public function getLogger()
+    {
+        if (!($this->logger instanceof LoggerInterface)) {
```
Comment (PavelLoparev) [inline inc/Smartling/Extensions/TranslateLock.php]: How is it possible to have logger with a type != `LoggerInterface`?

File: inc/Smartling/Extensions/TranslateLock.php [hunk 3]
```diff
@@ -26,22 +103,59 @@ public function getName()
     }
 
     /**
-     * @inheritdoc
+     * @param $id
+     *
+     * @return SubmissionEntity|null
      */
-    public function register()
+    private function getSubmissionById($id)
```
Comment (PavelLoparev) [inline inc/Smartling/Extensions/TranslateLock.php]: Seems like this is some general method and shouldn't be in this class. Maybe we need some `SubmissionHelper`?

File: inc/Smartling/Base/SmartlingCoreUploadTrait.php [hunk 4]
```diff
@@ -147,10 +147,65 @@ public function getXMLFiltered(SubmissionEntity $submission)
         }
     }
 
+    /**
+     * @param SubmissionEntity $submission
+     *
+     * @return array
+     */
+    private function readLockedTranslationFieldsBySubmission(SubmissionEntity $submission)
```
Comment (PavelLoparev) [inline inc/Smartling/Base/SmartlingCoreUploadTrait.php]: Trait `SmartlingCoreUploadTrait` is getting overflown by such unrelated methods. I think it should be in some helper class of directly in your new `TranslateLock` class.

File: inc/Smartling/Base/SmartlingCoreUploadTrait.php (hunk 1 above)
Comment (paveldhq) [reply inc/Smartling/Base/SmartlingCoreUploadTrait.php]: agree for static, but it should not be inside ArrayHelper

File: inc/Smartling/Extensions/TranslateLock.php (hunk 3 above)
Comment (paveldhq) [reply inc/Smartling/Extensions/TranslateLock.php]: don't think so

**Pull request merged**